### ISSUE TITLE
Mac won't build

### ISSUE THREAD

**Issue by timothystulman** [InvisibleWrench/FlutterMidiCommand#68]
First apologies if I'm missing something obvious. I'm a long-time macOS swift developer but am just starting in Flutter. I had some trouble with the pod file (I'm on an M1 apple computer) but I think I got it worked out. What's the minimum requirement for macOS? Is it macOS13 (ventura)? Does that mean that I have to update to the beta to run this? And would my users also need to be on macOS 13? 

`/Users/timothystulman/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_midi_command-0.4.8/macos/Classes/SwiftFlutterMidiCommandPlugin.swift:884:66: warning: string interpolation produces a debug description for an optional value; did you mean to make this explicit?
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ~~~~~~~^~~~~~~~~~~~~~~~~~~~
                                                                 String(describing:         )
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ~~~~~~~^~~~~~~~~~~~~~~~~~~~
                                                                                             ?? <#default value#>
/Users/timothystulman/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_midi_command-0.4.8/macos/Classes/SwiftFlutterMidiCommandPlugin.swift:884:66: warning: string interpolation produces a debug description for an optional value; did you mean to make this explicit?
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ~~~~~~~^~~~~~~~~~~~~~~~~~~~
                                                                 String(describing:         )
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ~~~~~~~^~~~~~~~~~~~~~~~~~~~
                                                                                             ?? <#default value#>
error: module name "" is not a valid identifier
error: module name "" is not a valid identifier
error: Module name "" is not a valid identifier (in target 'Runner' from project 'Runner')
Command SwiftDriver emitted errors but did not return a nonzero exit code to indicate failure
error: Module name "" is not a valid identifier (in target 'Runner' from project 'Runner')
Command SwiftDriver emitted errors but did not return a nonzero exit code to indicate failure
/Users/timothystulman/StudioProjects/midi_test7/macos/Flutter/GeneratedPluginRegistrant.swift:8:8: error: compiling for macOS 10.11, but module 'flutter_midi_command' has a minimum deployment target of macOS 10.13: /Users/timothystulman/StudioProjects/midi_test7/build/macos/Build/Products/Release/flutter_midi_command/flutter_midi_command.framework/Modules/flutter_midi_command.swiftmodule/arm64-apple-macos.swiftmodule
import flutter_midi_command
       ^
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Runner' from project 'Runner')
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Flutter Assemble' from project 'Runner')
** BUILD FAILED **

Building macOS application...                                           
Build process failed
timothystulman@MacBook-Pro-10 macos % 
`

**Comment by timothystulman** [comment]
I've made some progress... Apologies for my misunderstanding regarding the minimum version number. Of course it's 10.13 not 13.0. I'm still getting the following errors. Can anyone help with this? 

--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:arm64, id:00006000-001C31A00A38801E }
{ platform:macOS, arch:x86_64, id:00006000-001C31A00A38801E }
/Users/timothystulman/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_midi_command-0.4.8/macos/Classes/SwiftFlutterMidiCommandPlugin.swift:884:66: warning: string interpolation produces a debug description for an optional value; did you mean to make this explicit?
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ~~~~~~~^~~~~~~~~~~~~~~~~~~~
                                                                 String(describing:         )
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ~~~~~~~^~~~~~~~~~~~~~~~~~~~
                                                                                          ?? <#default value#>
error: module name "" is not a valid identifier
error: Module name "" is not a valid identifier (in target 'Runner' from project 'Runner')
Command SwiftDriver emitted errors but did not return a nonzero exit code to indicate failure
error: /Users/timothystulman/StudioProjects/midi_test9/build/macos/Build/Intermediates.noindex/Runner.build/Debug/Runner.build/Objects-normal/arm64/.swiftdoc: No such file or directory (in target 'Runner' from project 'Runner')
error: /Users/timothystulman/StudioProjects/midi_test9/build/macos/Build/Intermediates.noindex/Runner.build/Debug/Runner.build/Objects-normal/arm64/.abi.json: No such file or directory (in target 'Runner' from project 'Runner')
error: /Users/timothystulman/StudioProjects/midi_test9/build/macos/Build/Intermediates.noindex/Runner.build/Debug/Runner.build/Objects-normal/arm64/.swiftsourceinfo: No such file or directory (in target 'Runner' from project 'Runner')
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Runner' from project 'Runner')
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Flutter Assemble' from project 'Runner')
** BUILD FAILED **

Exception: Build process failed

**Comment by maks** [comment]
@timothystulman  welcome to Flutter development.

When you have issues with packages or plugins in Flutter, a good starting point is to try running the example app that is included in the git repo of almost all plugins.
I this case, on my M1 macmini I did:
```
git clone https://github.com/InvisibleWrench/FlutterMidiCommand 
cd FlutterMidiCommand/example
flutter pub get
flutter run -d macos lib/main.dart
```
I then got an error, which included:
```
Launching lib/main.dart on macOS in debug mode...
lib/main.dart:1
Warning: Podfile is out of date
  This can cause issues if your application depends on plugins that do not support iOS or macOS.
  See https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin-platforms for details.
  If you have local Podfile edits you would like to keep, see https://github.com/flutter/flutter/issues/45197 for instructions.
To regenerate the Podfile, run:

  rm macos/Podfile

--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:arm64, id:00008103-0006215A1AD1001E }
{ platform:macOS, arch:x86_64, id:00008103-0006215A1AD1001E }
error: the following command failed with exit code 0 but produced no further output
CompileSwift normal arm64 /Users/maks/work/FlutterMidiCommand/ios/Classes/SwiftFlutterMidiCommandPlugin.swift (in target 'flutter_midi_command' from project 'Pods')
ld: warning: ignoring file /Users/maks/work/FlutterMidiCommand/example/macos/Pods/FlutterMacOS/FlutterMacOS.framework/FlutterMacOS, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
```
So I followed the instructions and did `rm macos/Podfile` in the example dir and then tried running again:
```
maks@macmini example % flutter run -d macos lib/main.dart 
Launching lib/main.dart on macOS in debug mode...
Running pod install...                                             492ms
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:arm64, id:00008103-0006215A1AD1001E }
{ platform:macOS, arch:x86_64, id:00008103-0006215A1AD1001E }
/Users/maks/work/FlutterMidiCommand/ios/Classes/SwiftFlutterMidiCommandPlugin.swift:884:66: warning: string interpolation produces a debug description for an optional value; did you mean to make this explicit?
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ~~~~~~~^~~~~~~~~~~~~~~~~~~~
                                                                 String(describing:         )
        print("central did fail to connect state \(peripheral) \(error?.localizedDescription)")
                                                                 ~~~~~~~^~~~~~~~~~~~~~~~~~~~
                                                                                             ?? <#default value#>
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Runner' from project 'Runner')
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Flutter Assemble' from project 'Runner')
Building macOS application...                                           
Syncing files to device macOS...                                    37ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on macOS is available at: http://127.0.0.1:59039/KPXrfYliuC0=/
The Flutter DevTools debugger and profiler on macOS is available at:
http://127.0.0.1:9101?uri=http://127.0.0.1:59039/KPXrfYliuC0=/
```

And as you can see the example app built and ran (albeit with some warnings) and I could see the Flutter app ui on screen.
I plugged in my Akai Fire controller and it immediately showed up in the app:

![image](https://user-images.githubusercontent.com/71999/192080247-c852d221-af00-446f-98b1-e5d47f22b73c.png)

So my recommendation would be to start with the example app in this repo and then once you have it running, work out from there towards what you want to build for your app.

For reference, my config is:
```
maks@macmini example % flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.0, on macOS 12.6 21G115 darwin-arm, locale en-AU)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.69.2)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

• No issues found!
```

**Comment by mortenboye** [comment]
Closing this for now. Feel free to reopen if issue reappears.
